src/context/feature-flag.tsx

```diff
@@ -0,0 +1,46 @@
+import { useAdminStore } from "medusa-react"
+import React, { useContext, useEffect, useState } from "react"
+import { AccountContext } from "./account"
+
+export const defaultFeatureFlagContext: {
+  featureToggleList: Record<string, boolean>
+  isFeatureEnabled: (flag: string) => boolean
+} = {
+  featureToggleList: {},
+  isFeatureEnabled: function (flag): boolean {
+    return !!this.featureToggleList[flag]
+  },
+}
+
+export const FeatureFlagContext = React.createContext(defaultFeatureFlagContext)
+
+export const FeatureFlagProvider = ({ children }) => {
+  const { isLoggedIn } = useContext(AccountContext)
+
+  const [featureFlags, setFeatureFlags] = useState<string[]>([])
+
+  const { store, isFetching } = useAdminStore()
+
+  useEffect(() => {
+    if (isFetching || !store || !isLoggedIn) {
+      return
+    }
+
+    setFeatureFlags(store["feature_flags"])
+  }, [isFetching, store, isLoggedIn])
+
+  const featureToggleList = featureFlags.reduce(
```

Comment:
issue(blocking):

The implementation of this context does not take into account our revised backend implementation of the feature flags response which is:
```ts
[ { "key": "sales_channels", "value": true }, { "key": "multi-warehouse", "value": false} ]
```